cmd/start.go

```diff
@@ -233,3 +236,49 @@ func initializeMetrics(
 		netProvider,
 	)
 }
+
+func initializeDiagnostics(
+	ctx context.Context,
+	config *config.Config,
+	netProvider net.Provider,
+	stakeMonitor chain.StakeMonitor,
+) {
+	registry, isConfigured := diagnostics.Initialize(
+		config.Diagnostics.Port,
+	)
+	if !isConfigured {
+		logger.Infof("diagnostics are not configured")
+		return
+	}
+
+	registry.RegisterSource("peers", func() string {
```

Comment:
I think we should also extract this part to a separate function (`RegisterConnectedPeersSource`) and place it in `pkg/diagnostics/diagnostics.go` (just like we do for metrics). This way we can reuse this code from `keep-ecdsa` as well.

Reply:
I've extracted code into RegisterConnectedPeersSource and placed it into pkg/diagnostics/diagnostics.go *of keep-core* but this way it doesn't really help to reuse it from keep-ecdsa (unless I'm missing something). Or you meant moving it into pkg/diagnostics/diagnostics.go of keep-common? If this is the case, should I also move 'Initialize' function from pkg/diagnostics/diagnostics.go of keep-core to pkg/diagnostics/diagnostics.go of keep-common (and drop keep-core's pkg/diagnostics/diagnostics.go completely)?

Reply:
I mean `pkg/diagnostics/diagnostics.go` of `keep-core`. `keep-ecdsa` depends on `keep-core` directly so we can use that public function.